### PR TITLE
add explanatory text re: -only and -or-later

### PR DESCRIPTION
**Pull request by jlovejoy**
add explanatory text re: -only and -or-later delineation in standard license header (versus default text in full copy of license). proposed solution to #617

### CODE REVIEW

**Comment by jlovejoy** [comment]
@tieguy @theopenchainproject - want to have a look here?

**Comment by jlovejoy** [comment]
if people think this explains thing well enough, then will add to other GNU licenses

**Comment by jlovejoy** [comment]
have reached out to FSF to get their review as well

File: src/GPL-2.0-only.xml [hunk 1]
```diff
@@ -27,8 +27,11 @@
       </p>
     </standardLicenseHeader>
     <notes>
-      This license was released: June 1991 This refers to when this
-      GPL 2.0 only is being used (as opposed to GPLv2 or later).
+      This license was released: June 1991. This license identifier refers to when 
+      GPL-2.0-only is being used as opposed to GPL-2.0-or-later. The option to use either that
```
Comment (tieguy) [inline src/GPL-2.0-only.xml]: Can I suggest replacing "The option to use... " with "The license header of individual files will indicate which of 'only' or 'later' applies to that particular file." I _think_ that's the same intent, but if not, might want to clarify in a different way?

**Comment by shanecoughlan** [comment]
> @tieguy @theopenchainproject - want to have a look here?

Looks like a good solution to me.

**Comment by jlovejoy** [comment]
thanks @tieguy and @shanecoughlan - I've gotten some feedback from Donald at FSF as well, so just need to nail down that with him and make some edits here... hopefully soon!

**Comment by swinslow** [comment]
Hi @jlovejoy, this looks good to me! Does this incorporate the changes to the notes that you were incorporating based on your discussion with FSF? If so, I'm glad to go ahead and push the merge button.

File: src/GPL-2.0-only.xml [hunk 2]
```diff
@@ -27,11 +27,12 @@
...
-      GPL-2.0-only is being used as opposed to GPL-2.0-or-later. The option to use either that
-      version of the license only or to make it available under any later version of that license
-      is denoted in the standard license header. (The standard license header in the license 
-      itself includes the default "or (at your option) any later version" language.)
+      This license was released: June 1991. This license identifier refers to the choice 
+      to use the code under GPL-2.0-only, as distinguished from use of code under GPL-2.0-or-later
+      (i.e., GPL-2.0 or some later version such as GPL 3, a possible future GPL 4,
+      and so on). The license notice (usually at the head of the file) states which of these
+      applies to the code in the file.
+      <p>The example in the exhibit to the license shows the license notice for the "or later" approach.</p>
```
Comment (tieguy) [inline src/GPL-2.0-only.xml]: I assume there is no way for FSF to issue/approve an example that does not show the or-later approach?

Comment (swinslow) [reply src/GPL-2.0-only.xml]: Probably not before we freeze the release for 3.5... which we're hoping to do today / Monday.  =)  But I haven't spoken with them, so maybe @jlovejoy can weigh in here.

**Comment by jlovejoy** [comment]
@tieguy  - note, on the SPDX license pages, we show the correct standard header (in terms of reflecting the only or or-later option) but the standard license text exhibit has the or-later language. This is what was confusing people.

I've now updated the language again to tighten up a bit and call out the above fact more explicitly. I've also added the rest of the licenses this applies to.  I think we can go with this.

Also note: AGPL-1.0 and LGPL-3.0 don't have this issue, as they don't include the license notice exhibit

**Pull request merged**